src/App.tsx

```diff
@@ -1,24 +1,194 @@
+
+      default:
+        return todos;
+    }
+  }, [todos, filter]);
+
+  const addTodo = async (title: string) => {
+    try {
+      const newTodo = {
+        title,
+        userId: USER_ID,
+        completed: false,
+      };
+
+      setTempTodo({
+        id: 0,
+        ...newTodo,
+      });
+
+      const createdTodo = await createTodo(newTodo);
+
+      setTodos(prevTodos => [...prevTodos, createdTodo]);
+    } catch {
+      setError('Unable to add todo');
+    } finally {
+      setTempTodo(null);
+    }
+  };
+
+  const deleteTodo = async (todoId: number) => {
+    try {
+      setLoadingTodos(prevIds => [...prevIds, todoId]);
+      await removeTodo(todoId);
+      setTodos(prevTodos => prevTodos.filter(todo => todo.id !== todoId));
+    } catch {
+      setError('Unable to delete a todo');
+    } finally {
+      setLoadingTodos(prevIds => prevIds.filter(id => id !== todoId));
+    }
+  };
+
+  const updateTodoInfo = async (
+    todoId: number,
+    newTodoData: Partial<Pick<Todo, 'title' | 'completed'>>,
+  ) => {
+    try {
+      setLoadingTodos(prevIds => [...prevIds, todoId]);
+      const updatedTodo = await updateTodo(todoId, newTodoData);
+
+      setTodos(prevTodos => prevTodos.map(todo => {
+        if (todo.id !== todoId) {
+          return todo;
+        }
+
+        return updatedTodo;
+      }));
+    } catch {
+      setError('Unable to update a todo');
+    } finally {
+      setLoadingTodos(prevIds => prevIds.filter(id => id !== todoId));
+    }
+  };
+
+  const handleToggleButton = () => {
+    todos.forEach(async (todo) => {
+      await updateTodoInfo(
+        todo.id,
+        { completed: !isAllTodosCompleted },
+      );
+    });
+  };
+
+  const handleDeleteComplitedTodos = async () => {
+    const deletePromises = completedTodos.map(todo => deleteTodo(todo.id));
+
+    try {
+      await Promise.all(deletePromises);
+    } catch {
+      setError('Unable to delete todos');
+    }
+  };
+
   if (!USER_ID) {
     return <UserWarning />;
   }
 
   return (
-    <section className="section container">
-      <p className="title is-4">
-        Copy all you need from the prev task:
-        <br />
-        <a href="https://github.com/mate-academy/react_todo-app-add-and-delete#react-todo-app-add-and-delete">React Todo App - Add and Delete</a>
-      </p>
-
-      <p className="subtitle">Styles are already copied</p>
-    </section>
+    <div className="todoapp">
+      <h1 className="todoapp__title">todos</h1>
+
+      <div className="todoapp__content">
+        <header className="todoapp__header">
+
+          <button
+            type="button"
+            className={cn('todoapp__toggle-all', {
+              active: isAllTodosCompleted,
+            })}
+            aria-label="active"
+            onClick={handleToggleButton}
+          />
+          <TodoForm
+            setError={setError}
+            addTodo={addTodo}
+            tempTodo={tempTodo}
+          />
+        </header>
```

Comment:
let's create separate component for `header`

Reply:
Why did you ignore this comment :)?
Let's create a separate component `header`